src/main/api/2/studio-api-2.yaml

```diff
@@ -1100,6 +1100,89 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /users/me/properties:
+    get:
+      tags:
+        - users
+      summary: Retrieve properties for the current user
+      description: 'Required Permission: "LOGGED_IN"'
+      operationId: getUserProperties
+      parameters:
+        - name: siteId
+          schema:
+            type: string
+          in: query
+          description: The id of the site (defaults to global)
```

Comment:
Can we have an optional param to only get matching `property_key`s?

Reply:
@jdrossl this needs to allow the query of a single "key" within a site or across sites.

The DB is indexed on the "key" as well as "site".

I don't think this API should paginate.

Reply:
@rart `siteId` is optional, that is why it says that defaults to global

@sumerjabri all existing APIs use an empty `siteId` to reference the global repo, I'm not sure if changing that to mean all sites is a good idea. Maybe we can change it to be a keyword like `siteId: 'ANY'`

Reply:
@rart what's an example where you need to scan for a property that's across sites? (global properties will flow to the global (system) repo)

Reply:
A concrete example: a global preference that has a site override. We would get all and apply/prevail the current site if there are overrides.

Overall, I’m thinking around the flexibility of the API for different uses. Don’t know how open or closed we want it.

Reply:
What's an example of such a property? (When we open things up we incur a cost to implement the general, I need to see the use-case more concretely.)

Reply:
@rart ^

Reply:
@rart how are you planning handling global vs site specific? will the UI set global defaults for all users and then let them set site specific values? how will users go back to the global defaults? how would a user change the global default keeping the site specifics?

Reply:
An example: a prop that allows customising the widgets of my sidebar just for myself and wanting a particular override for one or more sites.

(Today, widgets are an xml file that applies to everyone but with user properties we could allow more personalised configs.) Same principle could go for Dashboards, etc (e.g. I want to add/remove a widget for my view of dashboard for all sites or for a given site.)

This may not be 3.2. Am I thinking too big?

I think for 3.2, mostly we’ll use this to move the stuff we keep in local browser storage to user props so they persist across terminals and browser data clearing. These are things like: language, UI theme(s), state of pages widgets (expanded/collapsed, folder I’m at, etc), current sidebar page I’m on, preview compatibility determination, etc.

Reply:
We will do a `*` for all sites down the road when needed. For now, not sending `siteId` (or "" or NULL) means global, otherwise, spec out the site.